.drone.yml

```diff
@@ -18,54 +14,84 @@ steps:
       registry: quay.io
       repo: quay.io/ukhomeofficedigital/hocs-toolbox
       tags:
-        - build_${DRONE_BUILD_NUMBER}
         - ${DRONE_COMMIT_SHA}
         - branch-${DRONE_COMMIT_BRANCH/\//_}
     environment:
       DOCKER_PASSWORD:
         from_secret: QUAY_ROBOT_TOKEN
       DOCKER_USERNAME: ukhomeofficedigital+hocs_quay_robot
+    when:
+      branch:
+        exclude:
+          - main
+    depends_on:
+      - clone
 
----
-
-# we're doing this as a promote so one can manually run it
-# normally this is defined in hocs-data's .drone.yml
+  - name: build & push latest
+    image: plugins/docker
+    settings:
+      registry: quay.io
+      repo: quay.io/ukhomeofficedigital/hocs-toolbox
+      tags:
+        - ${DRONE_COMMIT_SHA}
+        - latest
+    environment:
+      DOCKER_PASSWORD:
+        from_secret: QUAY_ROBOT_TOKEN
+      DOCKER_USERNAME: ukhomeofficedigital+hocs_quay_robot
+    when:
+      branch:
+        - main
+    depends_on:
+      - clone
 
+---
 kind: pipeline
 type: kubernetes
-name: teardown
+name: deployment
 
 trigger:
   event:
     - promote
+  target:
+    exclude:
+      - "*-prod"
+      - teardown
 
 steps:
-  - name: teardown gamma
+  - name: deploy to not-prod
     image: quay.io/ukhomeofficedigital/kd:v1.16.0
     commands:
       - cd kube
-      - bash ./teardown.sh
+      - ./deploy.sh
     environment:
-      ENVIRONMENT: hocs-gamma
+      ENVIRONMENT: ${DRONE_DEPLOY_TO}
       KUBE_TOKEN:
-        from_secret: hocs_toolbox_hocs_gamma
-    when:
-      target:
-        - teardown
-        - teardown-gamma
+        from_secret: hocs_toolbox_${DRONE_DEPLOY_TO}
+      KUBE_SERVER: https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
+      VERSION: ${VERSION}
+
+---
+# we're doing this as a promote so one can manually run it
+# normally this is defined in hocs-data's .drone.yml
+
+kind: pipeline
+type: kubernetes
+name: teardown
 
-  - name: teardown delta
+trigger:
+  event:
+    - promote
+  target:
+    - teardown
+
+steps:
+  - name: teardown non-mainline
     image: quay.io/ukhomeofficedigital/kd:v1.16.0
     commands:
       - cd kube
       - bash ./teardown.sh
     environment:
-      ENVIRONMENT: hocs-delta
+      ENVIRONMENT: ${ENVIRONMENT}
       KUBE_TOKEN:
-        from_secret: hocs_toolbox_hocs_delta
-    when:
-      target:
-        - teardown
-        - teardown-delta
-
-...
+        from_secret: hocs_toolbox_${DRONE_DEPLOY_TO/-/_}
```

Comment:
This'll always resolve to `hocs_toolbox_teardown`, is it meant to?

Reply:
Good catch, changed...